pkg/apis/numaplane/v1alpha1/shared_types.go

```diff
@@ -40,13 +44,25 @@ const (
 	UpgradeStrategyProgressive UpgradeStrategy = "Progressive"
 )
 
-type RolloutStrategy struct {
-	Progressive ProgressiveStrategy `json:"progressive,omitempty"`
+// PipelineTypeRolloutStrategy specifies the Rollout Strategy for fields shared by Pipeline and MonoVertex
+type PipelineTypeRolloutStrategy struct {
+	PipelineTypeProgressiveStrategy `json:"progressive,omitempty"`
 }
 
-// PipelineTypeRolloutStrategy specifies the RolloutStrategy for fields shared by Pipeline and MonoVertex
-type PipelineTypeRolloutStrategy struct {
-	RolloutStrategy `json:",inline"`
+// PipelineTypeProgressiveStrategy specifies the Progressive Rollout Strategy for fields shared by Pipeline and MonoVertex
+type PipelineTypeProgressiveStrategy struct {
+	Progressive ProgressiveStrategy `json:",inline"`
+
+	Analysis Analysis `json:"analysis,omitempty"`
+}
```

Comment:
My main motivation for this refactoring was to separate the `Analysis` section from the `ProgressiveStrategy` such that PipelineRollout and MonoVertexRollout would include `Analysis` but ISBServiceRollout would not.